### PR TITLE
Adds support for GOAL_PROGRESS and WEIGHTED_ELEMENTS complications.

### PR DESCRIPTION
**Pull request by garanj**
Updates Complication data source suite to add support for:

- GOAL_PROGRESS
- WEIGHTED_ELEMENTS

Data source types.

### CODE REVIEW

File: WearComplicationDataSourcesTestSuite/Wearable/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -20,10 +20,12 @@
         <string name="no_data_label">No Data</string>
         <string name="short_text_label">Short Text</string>
         <string name="ranged_value_label">Ranged Value</string>
+        <string name="goal_progress_label">Goal Progress</string>
         <string name="long_text_label">Long Text</string>
         <string name="icon_label">Icon</string>
         <string name="small_image_label">Small Image</string>
         <string name="large_image_label">Large Image</string>
+        <string name="weighted_elements">Large Image</string>
```
Comment (yschimke) [inline WearComplicationDataSourcesTestSuite/Wearable/src/main/res/values/strings.xml:28]:
```suggestion
        <string name="weighted_elements">Weighted Elements</string>
```

File: WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/ComplicationToggleReceiver.kt [hunk 2]
```diff
@@ -83,7 +84,12 @@ class ComplicationToggleReceiver : BroadcastReceiver() {
          * Returns the [ComplicationToggleArgs] from the [Intent] sent to the [ComplicationToggleArgs].
          */
         private fun Intent.getArgs(): ComplicationToggleArgs = requireNotNull(
-            extras?.getParcelable(EXTRA_ARGS),
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                extras?.getParcelable(EXTRA_ARGS, ComplicationToggleArgs::class.java)
+            } else {
+                @Suppress("DEPRECATION")
+                extras?.getParcelable(EXTRA_ARGS)
+            }
```
Comment (yschimke) [inline WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/ComplicationToggleReceiver.kt:92]:
```suggestion
            BundleCompat.getParcelable(extra, EXTRA_ARGS, ComplicationToggleArgs::class.java)
```

File: WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/GoalProgressDataSourceService.kt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ * A complication provider that supports only [ComplicationType.GOAL_PROGRESS] and cycles
+ * through the possible configurations on tap. The value is randomised on each update.
+ *
+ * Note: This subclasses [SuspendingComplicationDataSourceService] instead of [ComplicationDataSourceService] to support
+ * coroutines, so data operations (specifically, calls to [DataStore]) can be supported directly in the
+ * [onComplicationRequest].
+ *
+ * If you don't perform any suspending operations to update your complications, you can subclass
+ * [ComplicationDataSourceService] and override [onComplicationRequest] directly.
+ * (see [NoDataDataSourceService] for an example)
```
Comment (yschimke) [inline WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/GoalProgressDataSourceService.kt:44]: I'm not sure we should give this advice, it mainly creates confusion. Shouldn't the suspending perform be preferred as a standard API, and better base for future changes.

Comment (garanj) [reply WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/GoalProgressDataSourceService.kt:44]: Agree. Copied this directly over... I'll update this and the other existing complications.

File: WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/GoalProgressDataSourceService.kt [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+ */
+@RequiresApi(33)
+class GoalProgressDataSourceService : SuspendingComplicationDataSourceService() {
+
+    override suspend fun onComplicationRequest(request: ComplicationRequest): ComplicationData? {
+        if (request.complicationType != ComplicationType.GOAL_PROGRESS) {
+            return null
```
Comment (yschimke) [inline WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/GoalProgressDataSourceService.kt:51]: null means no change to data?  Should this be NoDataComplicationData?

https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:wear/watchface/watchface-complications-data-source-ktx/src/main/java/androidx/wear/watchface/complications/datasource/SuspendingComplicationDataSourceService.kt?q=SuspendingComplicationDataSourceService

https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:wear/watchface/watchface-complications-data/src/main/java/androidx/wear/watchface/complications/data/Data.kt;l=317

```
/**
 * Type that can be sent by any complication data source, regardless of the configured type, when
 * the complication data source has no data to be displayed. If no [placeholder] is included then
 * watch faces may choose whether to render this in some way or leave the slot empty.
 ```

**Review by yschimke — APPROVED** [review]

**Pull request merged**